modules/unzip/main.nf

```diff
@@ -0,0 +1,40 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process UNZIP {
+    tag '$bam'
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:[:], publish_by_meta:[]) }
+
+
+    conda (params.enable_conda ? "bioconda::p7zip=15.09" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/p7zip:15.09--h2d50403_4"
+    } else {
+        container "quay.io/biocontainers/p7zip:15.09--h2d50403_4"
+    }
+
+    input:
+    path archive
+
+    output:
+    path "*"             , emit: result
```

Comment:
this will also emit the `version.txt`. 
Maybe best to unzip to a subdirectory and only return its contents?